eng/generators.targets

```diff
@@ -41,17 +41,18 @@
        That is required as the EnabledGenerators condition checks on the Reference and ProjectReference items and hence can't be a property condition. -->
   <ItemGroup Condition="'@(EnabledGenerators)' != '' and
                         @(EnabledGenerators->AnyHaveMetadataValue('Identity', 'LibraryImportGenerator'))">
-    <ProjectReference Include="$(LibrariesProjectRoot)System.Runtime.InteropServices\gen\LibraryImportGenerator\LibraryImportGenerator.csproj;
-                                $(LibrariesProjectRoot)System.Runtime.InteropServices\gen\Microsoft.Interop.SourceGeneration\Microsoft.Interop.SourceGeneration.csproj"
-                      OutputItemType="Analyzer"
-                      ReferenceOutputAssembly="false" />
+    <AnalyzerReference Include="$(LibrariesProjectRoot)System.Runtime.InteropServices\gen\LibraryImportGenerator\LibraryImportGenerator.csproj;
```

Comment:
Do we still need to do this in targets?  Could we make libraries in the shared framework use `AnalyzerReference` just like they have to declare their reference to Interop and then have libraries outside the shared framework just get it (like customer projects will, now that its in the refpack)?

Reply:
That's eventually the goal, yes. I just yesterday discussed this with @jkoritzinsky in the infrastructure stand-up. Anything that's OOB should get the analyzers via the targeting pack and anything inside the shared framework should explicitly reference the analyzers. Devil in the details: You still need an explicit analyzer reference when you build pre net7.0 (net6.0, netstandard2.0, net462).
Therefore, we will still need to reference the analyzer for anything that is not NetCoreAppCurrent. In addition to that, Visual Studio's CPS currently has a bug that causes ProjectReferences that aren't referenced by the "main" active configuration to not be built. Because of that, I would prefer to keep the explicit analyzer reference for now.

Reply:
> Do we still need to do this in targets? Could we make libraries in the shared framework use AnalyzerReference just like they have to declare their reference to Interop

Just to check, you did see that this logic doesn't run inside a target, just in a targets file, right?

What you are proposing could definitely be done BUT the current logic in generators.targets already does that automatically when you either reference System.Runtime.InteropServices or System.Private.CoreLib. IMO, I would treat the `LibraryImportGenerator` source generator as a transitive analyzer dependency of InteropServices and CoreLib. The current logic already does that.

Reply:
> Just to check, you did see that this logic doesn't run inside a target, just in a targets file, right?

Yes.  I was just thinking that if we made it easy enough for projects to opt in it might be better to have them explicitly opt into referencing the analyzer, just like they explicitly reference every other reference.  It could avoid that magic/complexity in the condition, and it  might reduce the number of places that actually reference the analyzer.